core/kernel/src/feature_manifest.rs

```diff
@@ -0,0 +1,270 @@
+use serde::Deserialize;
+use std::collections::HashMap;
+
+#[derive(Deserialize, PartialEq, Debug)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)] // Temporary until exposed through C API
+pub struct FeatureManifest {
+    name: Option<String>,
+    author: Option<String>,
+    version: Option<String>,
+    description: Option<String>,
+    thumbnail: Option<String>,
+    thumbnail_color: Option<String>,
+    primary_color: Option<String>,
+    links: Option<HashMap<String, String>>,
+}
+
+#[derive(PartialEq, Debug)]
+pub struct FeatureManifestParsingError {
+    description: String,
+}
+
+// Alias for str, probably will be moved to a centralized place to be reused
+pub type JSONStr = str;
+
+#[allow(dead_code)] // Temporary until exposed through C API
+impl FeatureManifest {
+    pub fn parse(
+        json: &JSONStr,
+        locale: &str,
```

Comment:
I'm not totally sure that what you want here is the locale. IIRC, it's the translation key what you are going to be using.  It's derived from the locale (again, see #690)  but not the same.

Reply:
We would use locale's language code, I will update `locale` to `language_code` to be more specific.

Reply:
Please bear in mind that, in general, this will not be an environment variable, but something that needs to be extracted from the OS. That's already done in the JS shared code, wonder if there would be a way of integrating them (and not duplicating code)

Reply:
Exactly my reasoning - this variable would be injected by the platform based on its locale API.